### PR TITLE
Refactor result

### PR DESCRIPTION
**Pull request by oscarbc96**
- Refactor Result class
  - Deprecate warning in favour of Monitor (updates one rule)
  - Updates lots of tests

### CODE REVIEW

File: cfripper/rules/wildcard_resource_rule.py [hunk 1]
```diff
@@ -103,15 +103,15 @@ def _add_to_result(
         action: Optional[str],
         statement: Statement,
         extras: Dict,
-        warning: bool = False,
+        monitor: bool = False,
```
Comment (oscarbc96) [inline cfripper/rules/wildcard_resource_rule.py:106]: Warning has been removed, use monitor instead

File: tests/utils.py [hunk 2]
```diff
@@ -33,10 +34,7 @@ def get_cfmodel_from(path: str) -> CFModel:
...
 
-def compare_lists_of_failures(list_1: List, list_2: List) -> bool:
-    def get_reason(item):
-        return item["reason"] if isinstance(item, dict) else item.reason
-
-    return len(list_1) == len(list_2) and sorted(list_1, key=get_reason, reverse=True) == sorted(
-        list_2, key=get_reason, reverse=True
+def compare_lists_of_failures(list_1: List[Failure], list_2: List[Failure]) -> bool:
+    return len(list_1) == len(list_2) and sorted(list_1, key=lambda item: item.reason, reverse=True) == sorted(
+        list_2, key=lambda item: item.reason, reverse=True
```
Comment (oscarbc96) [inline tests/utils.py:39]: Compare just Filter objects, not dicts and objects

File: cfripper/model/result.py [hunk 3]
```diff
@@ -32,105 +32,53 @@ def serializable(self):
...
-
-        # Update the attribute dict with the properties
-        if props:
-            attribs.update({prop: getattr(self, prop) for prop in props})
-        return attribs
-
-    def __repr_args__(self):
-        return self.dict().items()  # type: ignore
-
-    # End of temporary fix
```
Comment (oscarbc96) [inline cfripper/model/result.py:83]: this is no longer needed

File: cfripper/model/result.py [hunk 4]
```diff
@@ -32,105 +32,53 @@ def serializable(self):
 class Result(BaseModel):
     class Config(BaseModel.Config):
         extra = Extra.forbid
-
-    failed_rules: List[Failure] = []
-    exceptions: List = []
-    failed_monitored_rules: List[Failure] = []
-    warnings: List[Failure] = []
```
Comment (oscarbc96) [inline cfripper/model/result.py:39]: Merge all failures into one list

**Review by ocrawford555 — APPROVED** [review]
Great work! Looks like a lot simpler code.

Will this also close/fix this issue: https://github.com/Skyscanner/cfripper/issues/103?

**Review by ignaciobolonio — APPROVED** [review]
Much cleaner 😄

**Comment by oscarbc96** [comment]
@ocrawford555 yep, but I think we could refactor a little more in a following PR

**Pull request merged**